### PR TITLE
Feature/fix performance issue

### PR DESCRIPTION
**Pull request by Norbi1986**
### Summary
We noticed a performance issue with the latest release(1.0.9). 
We did a root cause analysis and we found the source of issue.
The "[os.path.join(self.directory, fname) for fname in self.filenames]" was refactorized into @property function and had been called with each iteartion by _get_batches_of_transformed_samples() function:
e.g: 100(iteration) x 155.096(number of iamges) = 15.509.600(path generated)

I moved the filepath generator to the _init_() function from the @property function
So, the file_paths will be generated just one time with _init()_ function

### PR Overview

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [ ] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by datumbox** [comment]
This patch improves significantly the speed. From 1.0.5 to 1.0.9 the execution times increased by 5% due to the refactoring. This patch reduced the execution times comparing to 1.0.9 by 12%.

File: keras_preprocessing/image/dataframe_iterator.py [hunk 1]
```diff
@@ -161,6 +161,10 @@ def __init__(self,
                                                 batch_size,
                                                 shuffle,
                                                 seed)
+        root = self.directory or ''
+        self.path_to_files = [
```
Comment (rragundez) [inline keras_preprocessing/image/dataframe_iterator.py]: Can you call the attribute `self.filepaths`? It is a common practice to add a prefix underscore if the attribute will be exposed via a property, thanks. 
Perhaps we do not need to create that intermediate `root` variable.
and can you change this logic above the instantiation of the parent class? Line 159

File: keras_preprocessing/image/directory_iterator.py [hunk 2]
```diff
@@ -140,9 +140,13 @@ def __init__(self,
...
                                                 seed)
 
+        self.path_to_files = [
+            os.path.join(self.directory, fname) for fname in self.filenames
+        ]
+
     @property
     def filepaths(self):
-        return [os.path.join(self.directory, fname) for fname in self.filenames]
+        return self.path_to_files
```
Comment (rragundez) [inline keras_preprocessing/image/directory_iterator.py]: Can you call the attribute `self.filepaths`? It is a common practice to add a prefix underscore if the attribute will be exposed via a property, thanks. 
Perhaps we do not need to create that intermediate `root` variable.
and can you change this logic above the instantiation of the parent class? Line 138

File: keras_preprocessing/image/dataframe_iterator.py (hunk 1 above)
Comment (Norbi1986) [reply keras_preprocessing/image/dataframe_iterator.py]: Yes, I fixed my code according to your remark

File: keras_preprocessing/image/directory_iterator.py (hunk 2 above)
Comment (Norbi1986) [reply keras_preprocessing/image/directory_iterator.py]: Yes, I fixed my code according to your remark

File: keras_preprocessing/image/directory_iterator.py [hunk 3]
```diff
@@ -90,7 +90,7 @@ def __init__(self,
                                                             save_format,
                                                             subset,
                                                             interpolation)
-        self.directory = directory
+        self.directory = directory or ''
```
Comment (rragundez) [inline keras_preprocessing/image/directory_iterator.py]: In `DirectoryIterator` a the `directory` argument is not optional, no need for the `or` operator.

Comment (Norbi1986) [reply keras_preprocessing/image/directory_iterator.py]: Yes, you are right.
I fixed it

**Comment by rragundez** [comment]
Thanks for your contribution! LGTM

**Pull request merged**